feat: Parkour post (#11)
Alystrasz/NorthstarBlog: _posts/2024-01-08-parkour-tournament.md

---
title: "Birthday's Parkour tournament"
header:
  image: /assets/images/page-header-image.jpg
  og_image: /assets/images/page-header-image.jpg
tags:
  - FNN
author: Alystrasz
last_modified_at: 2024-01-21T13:09:00-01:00
---

Happy second anniversary, Northstar !

To celebrate two years of existence, we decided to organize a competition between all players to see who's the fastest:

> this blog post tells the story of the Parkour mod's origin and development, and showcases results of this first Parkour

Author: Why capitalizing "this" while it's in the middle of a sentence?

> this blog post tells the story of the Parkour mod's origin and development, and showcases results of this first Parkour

Reviewer: I thought the word after a colon `:` is capitalised or am I mistaken? ^^"

> this blog post tells the story of the Parkour mod's origin and development, and showcases results of this first Parkour

Author: In English, it seems like it depends, in French you never capitalize after a color except if it's a proper noun :)

development, and showcases results of this first Parkour tournament.

### Developer story

#### Origin of the mod

> When I was younger, I loved going to my friends' houses, set up a local network, and play some LAN games: these games

Reviewer: Doing `.` instead of colon feels more natural to me ^^

```suggestion
When I was younger, I loved going to my friends' houses, set up a local network, and play some LAN games. These games
```

Author: Addressed in 7335657c915032099b3354c7cb53c20c64d53d06.

included Counter Strike 1.6, Warcraft 3, and [Trackmania](https://www.trackmania.com/?lang=en); if you don't know the
last, it's a car game where you have to reach a finish line, crossing a set of checkpoints as fast as you can (rings a
bell, right?).

> I also spend lots of (probably more than I should) time on World of Warcraft, which latest expansion, named Dragonflight,

Reviewer: Grammar wise, I think this should `whose` (https://english.stackexchange.com/questions/429976/possessive-form-of-which)
```suggestion
I also spend lots of (probably more than I should) time on World of Warcraft, whose latest expansion, named Dragonflight,
```

Author: Addressed in 7335657c915032099b3354c7cb53c20c64d53d06.

> released a similar feature dubbed ["Dragonriding races"](https://www.youtube.com/watch?v=7Gr4rZXPw6w).

Reviewer: (off-topic)

The video shows a waypoint indicator for the next checkpoint, maybe we should do something like that for Parkour as well :clueless:

Author: We definitely should have a way to draw beautiful lines to do that, yep!

> At some point, I thought: "that would be a great mod idea for Northstar!". I would call it *"Parkour"*.

Reviewer: I think grammar wise this should be capitalised but not sure.

```suggestion
At some point, I thought: "That would be a great mod idea for Northstar!". I would call it *"Parkour"*.
```

> <img src="https://j.gifs.com/zKxZPy.gif" />

Reviewer: Thinking this should maybe be copied and an internal link. Just to reduce the reliance on external services ^^

Author: There's a tradeoff between this and reducing local storage I guess :)

Reviewer: I think we can handle one extra file :P

Author: Addressed in 2499222920680a667e1f4dd227827488b03a08b5.

#### Development

Since developing this mod basically resolved to develop a multiplayer version of the single-player 
[Gauntlet](https://titanfall.fandom.com/wiki/The_Pilot's_Gauntlet) mission, I could reuse a lot of assets from it,
including starting/finish lines visuals, speed indicator and scoreboards.
For other entities I had to improvise: for instance, the model used for checkpoints is actually a Titan shield, colored

> in green :)

Reviewer: Could do a screenshot or GIF of the checkpoint bubble here ^^

Sidenote, [gifski](https://gif.ski/) is an awesome tool to convert videos to high quality GIFs ^^

Author: Addressed in 52a447101a29de3d51829522b77142ec72700b6e.

Initial development of the mod only included a local scoreboard, where you could see scores of players in the current

> match; we then thought it would be cool to have a way to save your personal best time for each map, which led me to

Reviewer: Something about the grammar in this sentence feels odd but not sure how to best phrase it ^^

develop a scoring server to save scores for all players in all maps, and an associated web scoreboard, to maintain
competition between players:

<img src="{{ 'assets/images/posts/parkour/global-scoreboard.png' | relative_url }}" alt="Screenshot of the Parkour tournament web scoreboard" />

Once the scoring server was set up, I deployed Parkour servers in Europe, Americas and Australia during twelve days,

> which led to X players scoring a time on a map (maybe more tried, but couldn't finish a route? We'll never know).

Reviewer: I'm assuming the `X` is placeholder? ^^

Author: Exactly yep, will be replaces in two hours when I shutdown all servers

Author: Addressed in ca192a61006c11d14d1a17dfd005fd4291d463e5.

> Once the scoring server was set up, I deployed Parkour servers in Europe, Americas and Australia during twelve days,

Reviewer: ```suggestion
Once the scoring server was set up, I deployed Parkour servers in Europe, North America, and Australia during twelve days,
```

Don't think we had a South American server ^^"

Author: Addressed in 7335657c915032099b3354c7cb53c20c64d53d06.

#### Feedback

We received *a lot* of feedback about Parkour which gave us lots of ideas to improve it for the next tournament, and I

> thank you all for that.

Reviewer: Could put some examples here together with who suggested it to give a bit of a sense of community.
(Although that's quite a bit of extra work so I understand if you wanna skip that :P)

Author: I wanna keep some surprise for the next release :)

This Parkour tournament was also a good opportunity to test *mod auto-downloading* in live conditions, if you're curious
about this feature, I wrote [another blog post](https://remyraes.com/autodownload.html) describing it in details.

### Media

To celebrate Northstar's second birthday and the parkour tournament, we asked the community some help with trailers, and
got an amazing one from [P3NG00N](https://www.youtube.com/@P3NG00Nlol):

<iframe width="560" height="315" src="https://www.youtube-nocookie.com/embed/1DSNwdV8ahA" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>

> ---

Reviewer: `<hr>` in Markdown is `***` IIRC
```suggestion
***
```

Author: Both are valid according to the [Markdown guide](https://www.markdownguide.org/basic-syntax/#horizontal-rules).

Reviewer: Huh, TIL. I'd still go with `***` though cause GitHub web diff view treats `---` differently and renders the text after it as YAML ^^

![image](https://github.com/Alystrasz/NorthstarBlog/assets/40122905/5bc12007-5ae5-46ab-85fb-86e5a93567ed)

> But we were also lucky enough to have [Evan Boymel](https://www.youtube.com/@evanboymelviper), voice actor of Viper,

Reviewer: I would drop the `but`, sounds nicer ^^
```suggestion
We were also lucky enough to have [Evan Boymel](https://www.youtube.com/@evanboymelviper), voice actor of Viper,
```

<br/>
Associated YouTube short from Evan:

<iframe width="560" height="560" src="https://www.youtube-nocookie.com/embed/etnCR3pHPBY" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>

### Closing words

I was really surprised seeing the Titanfall 2 speedrunning community, including [Dadadu2711](https://www.twitch.tv/videos/2028529918),
taking over the mod and breaking the routes I designed in ways I never heard of previously: grav star boosting, EPG
boosting... There were discussions about the best ways to improve scores; I feel like there was a good competitive
spirit within this tournament, that's definitely something I plan to redo in the future!

I want to thank all players that participated in that first tournament, and all developers + play testers who helped
make this mod what it is today.

> See you on the Frontier!

Reviewer: ```suggestion
See you all on the Frontier!
```

Sounds a bit more inclusive ^^

Author: Addressed in 7335657c915032099b3354c7cb53c20c64d53d06.
